bootstrap/main.zig

```diff
@@ -0,0 +1,59 @@
+const std = @import("std");
+const logger = @import("logger");
+const sdl3 = @import("sdl3");
+const build_options = @import("options").build_options;
+
+const log = logger.Scoped(.main);
+
+const DedicatedBootstrapper = @import("DedicatedBootstrapper.zig");
+
+pub fn main() !void {
+    var debug_allocator_impl: std.heap.DebugAllocator(.{
+        .stack_trace_frames = 10,
+    }) = .init;
+    defer if (build_options.safety and debug_allocator_impl.deinit() == .leak) {
+        @panic("Memory leak!");
+    };
+
+    const gpa = debug_allocator_impl.allocator();
```

Comment:
We should be using the smp_allocator in unsafe build modes.

```suggestion
    var debug_allocator_impl: std.heap.DebugAllocator(.{
        .stack_trace_frames = 10,
    }) = .init;
    defer if (build_options.safety and debug_allocator_impl.deinit() == .leak) {
        @panic("Memory leak!");
    };

    const gpa = if (build_options.safety) debug_allocator_impl.allocator() else std.heap.smp_allocator;
```